cli/install/install.go

```diff
@@ -0,0 +1,654 @@
+}
+
+// detectOsName returns name of the OS.
+func detectOsName() string {
+	if runtime.GOOS == "darwin" {
+		return "darwin"
+	}
+	if runtime.GOOS == "windows" {
+		return "windows"
+	}
+	if runtime.GOOS == "linux" {
+		distroInfo := getDistroInfo()
+		return distroInfo.Name
+	}
+	return ""
+}
+
+// getLatestTarantoolTag returns latest tag from tarantool repository.
+func getLatestTarantoolTag() (string, error) {
+	// TODO Use version functions from util/version (will be implemented later)
+
+	// Get all tags from tarantool git repo.
+	cmd := exec.Command("git", "ls-remote", "--tags", "--sort="+
+		"v:refname", TarantoolLink)
+	readPipe, writePipe, _ := os.Pipe()
+	cmd.Stdout = writePipe
+	cmd.Stderr = os.Stderr
+	cmd.Start()
+	err := cmd.Wait()
+	writePipe.Close()
+	if err != nil {
+		return "", err
+	}
+
+	var buf bytes.Buffer
+	io.Copy(&buf, readPipe)
+	versions := buf.String()
+	versionsArray := strings.Split(versions, "\n")
+	if len(versionsArray) == 0 {
+		return "", fmt.Errorf("Could not get latest version of tarantool.")
+	}
+	// Get last tag.
+	latestTag := versionsArray[len(versionsArray)-2]
+	latestTag = strings.TrimSuffix(latestTag, "^{}")
+	trimpos := strings.LastIndex(latestTag, "/") + 1
+	latestTag = latestTag[trimpos:]
+	return latestTag, nil
+}
+
+// checkDependecie checks if programm is installed.
+func checkDependecie(programm string) bool {
+	cmd := exec.Command(programm, "--version")
+	cmd.Start()
+	err := cmd.Wait()
+	return err == nil
+}
+
+// checkPythonDependencie checks if python packet is installed.
+func checkPythonDependencie(packet string) bool {
+	cmd := exec.Command("pip3", "list")
+	readPipe, writePipe, _ := os.Pipe()
+	cmd.Stdout = writePipe
+	cmd.Stderr = os.Stderr
+	cmd.Start()
+	cmd.Wait()
+	writePipe.Close()
+	var buf bytes.Buffer
+	io.Copy(&buf, readPipe)
+	packets := buf.String()
+	return strings.Contains(packets, packet)
+}
+
+// checkDeianLibDependencie checks if library is installed on Debian/Ubuntu.
+func checkDebianLibDependencie(library string) bool {
+	cmd := exec.Command("dpkg", "-L", library)
+	cmd.Start()
+	if cmd.Wait() == nil {
+		return true
+	} else {
+		return false
+	}
+}
+
+// checkLibDependencie checks if library is installed
+func checkLibDependencie(library string) bool {
+	cmd := exec.Command("")
+	if strings.Contains(detectOsName(), "Ubuntu") || strings.Contains(detectOsName(), "Debian") {
+		return checkDebianLibDependencie(library)
+	}
+	if strings.Contains(detectOsName(), "darwin") {
+		cmd = exec.Command("brew", "list")
+	}
+	if strings.Contains(detectOsName(), "CentOS") {
+		cmd = exec.Command("yum", "list", "--installed")
+	}
+	readPipe, writePipe, _ := os.Pipe()
+	cmd.Stdout = writePipe
+	cmd.Stderr = os.Stderr
+	cmd.Start()
+	cmd.Wait()
+	writePipe.Close()
+	var buf bytes.Buffer
+	io.Copy(&buf, readPipe)
+	packets := buf.String()
+	return strings.Contains(packets, library)
+}
+
+// checkTtDependencies checks if tt dependencies is installed.
+func checkTtDependencies() bool {
```

Comment:
I'm not sure, that these binaries are tt dependencies. Are they specific command deps? I think I can run tt without git or python3

Reply:
Git is prerequisite of tt, and also runtime dependencie of install and search. Python is needed for test purposes. If user doesn't want to install them, he can use install -f to skip that check.